experimenter/experimenter/experiments/models.py

```diff
@@ -1347,6 +1353,124 @@ def get_branch_data(self, analysis_basis, selected_segment):
+            else {}
+        )
+
+        for branch in branches:
+            branch_slug = branch.slug
+            branch_results = (
+                overall_results.get(branch_slug, {}).get("branch_data", {})
+                if isinstance(overall_results, dict)
+                else {}
+            )
+            branch_metrics = {}
+
+            for metric in metrics:
+                slug = metric.get("slug")
+                group = metric.get("group")
+
+                metric_src = branch_results.get(group, {}).get(slug, {})
+
+                absolute_data_list = metric_src.get("absolute", {}).get("all", [])
+                relative_data_list = (
+                    metric_src.get("relative_uplift", {})
+                    .get(reference_branch, {})
+                    .get("all", [])
+                )
+
+                absolute_data_list.sort(key=self.formattedAnalysisPointComparator)
+                relative_data_list.sort(key=self.formattedAnalysisPointComparator)
+
+                significance_map = (
+                    metric_src.get("significance", {})
+                    .get(reference_branch, {})
+                    .get(window, {})
+                )
+
+                abs_entries = []
+                for i, data_point in enumerate(absolute_data_list):
+                    lower = round(data_point.get("lower"), 2)
+                    upper = round(data_point.get("upper"), 2)
+
+                    if metric.get("display_type") == "percentage":
+                        lower = f"{lower * 100}%"
+                        upper = f"{upper * 100}%"
+
+                    significance = significance_map.get(str(i + 1), "neutral")
+                    abs_entries.append(
+                        {"lower": lower, "upper": upper, "significance": significance}
+                    )
+
+                rel_entries = []
+                for i, data_point in enumerate(relative_data_list):
+                    lower = data_point.get("lower")
+                    upper = data_point.get("upper")
+                    interval = abs(round((lower + upper) / 2 * 100))
+                    significance = significance_map.get(str(i + 1), "neutral")
+                    rel_entries.append(
+                        {
+                            "lower": lower,
+                            "upper": upper,
+                            "significance": significance,
+                            "interval": interval,
+                        }
+                    )
+
+                branch_metrics[slug] = {"absolute": abs_entries, "relative": rel_entries}
+
+            metric_data[branch_slug] = branch_metrics
+
+        return metric_area_data
+
+    def get_metric_data(
+        self, analysis_basis, segment, reference_branch, window="overall"
+    ):
+        metric_areas = self.metric_areas
+        metric_data = {}
+
+        for area in metric_areas:
+            match area:
+                case "KPI Metrics":
```

Comment:
String constants like this should always be declared somewhere as a constant in a consistent place, otherwise you have to go hunting around for random possibly duplicated strings.